discord/utils.py

```diff
@@ -339,12 +374,11 @@ def deprecated(
     since: str | None = None,
     removed: str | None = None,
     reference: str | None = None,
-    stacklevel: int = 3,
+    stacklevel: int = 1,
     *,
     use_qualname: bool = True,
-) -> Callable[[Callable[[P], T]], Callable[[P], T]]:
-    """A decorator implementation of :func:`warn_deprecated`. This will automatically call :func:`warn_deprecated` when
-    the decorated function is called.
+) -> _typing_deprecated:
```

Comment:
I should test if this is actually detected correctly by type checkers

Reply:
And?

Reply:
I forgot to check

Reply:
So it is not apparently, at least basedpyright. But in general imo this solution in this pr was kinda hacky. The thing is to solve #1917 we either have to give up the custom message thing, or give up it being picked up by type checkers.

Additionally, things like `cached_slot_property` make it not be picked up by type checkers too.